### DISCUSSION TITLE
Diagonal Mirror symmetries?

### DISCUSSION THREAD

**Discussion by Andeloth** [NanoComp/meep#2410] [Q&A] [answered]
Do meep simulations support mirror symmetries along a non-grid direction? I.e if I wanted to have a diagonal across which my simulation is symmetric, how would I do it? Do I have to effectively rotate the whole simulation 45 degrees and then use an X or Y symmetry, or can I specify a non-grid direction for a Mirror object?

An example is @smartalecH 's paper here https://opg.optica.org/oe/fulltext.cfm?uri=oe-30-17-31058&id=490977 .

**Comment by smartalecH** [comment, marked as answer]
> Do meep simulations support mirror symmetries along a non-grid direction?

[No](https://meep.readthedocs.io/en/latest/Exploiting_Symmetry/) unfortunately. That would be rather tricky...

> Do I have to effectively rotate the whole simulation 45 degrees and then use an X or Y symmetry

Even if you did this (not a bad idea) you still may not be able to leverage a *simulation* symmetry where the *fields* (which include *sources* and *monitors*) must also be symmetric.

In the case of the grating coupler you reference, we had this issue. We were injecting a *polarized* gaussian beam source and maximizing the power in *one* port. So the fields weren't symmetric. But we implicitly enforced a *geometric* symmetry constraint to ensure that the other polarization would couple into the other port. The fact that the polarization extinction was so good (40dB!) was a bit of luck tbh... we were also leveraging the fact that it's hard to "rotate" the polarization so quickly.

You *could* drive it with a gaussian beam polarized at 45 degrees (or 0 degrees if you rotate your structure) and maximize the power in both output arms simultaneously (just like we do with the 50-50 splitter tutorial). The challenge with this approach is I don't think you'll get the behavior you're looking for. Namely, you want a polarization demultiplexing GC. If you optimize based on an input superposition of states, the optimizer won't know if it's sufficiently mapping each polarization to the corresponding output arm.

To get around this, you could do a minimax over both "mixed" polarization states (45 degrees and -45 degrees). But then you're really not saving much computationally (technically it's still cheaper, because embarrassingly parallel simulations will always scale better than larger simulation distributed via MPI due to communication overhead... but is it worth all the extra software development time?)

Of course, I'm not sure what you're actually trying to simulate/design, so the above may not apply at all.

**Reply by Andeloth** [reply]
Thanks for the insight Alec, that's exactly what I was trying to figure out.

I'm trying to replicate your methods in that paper right now, and like 30 minutes ago I also realized that the fields won't be symmetric so we can only apply symmetries on the design parameters and not the whole simulation. Regardless, applying the parameter symmetry lets you use 1 simulation instead of 2 because you don't need to have the optimizer figure out how to split the 2 polarizations, you just make it create a grating that will do that by constraint.

Still, it's good to know that symmetries can only be applied along grid directions. The 45 degree rotation could be used to design 50-50 splitter grating couplers, or similar devices when your outputs are along different axes.
